// @redux-devtools/extension/lib/index.js
import { compose } from 'redux';
function extensionComposeStub(...funcs) {
    if (funcs.length === 0)
        return undefined;
    if (typeof funcs[0] === 'object')
        return compose;
    return compose(...funcs);
}
export const composeWithDevTools = typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : extensionComposeStub;
export const devToolsEnhancer = typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__
    ? window.__REDUX_DEVTOOLS_EXTENSION__
    : function () {
        return function (noop) {
            return noop;
        };
    };
export { composeWithDevTools as composeWithDevToolsDevelopmentOnly, devToolsEnhancer as devToolsEnhancerDevelopmentOnly, } from './developmentOnly.js';
export { composeWithDevTools as composeWithDevToolsLogOnly, devToolsEnhancer as devToolsEnhancerLogOnly, } from './logOnly.js';
export { composeWithDevTools as composeWithDevToolsLogOnlyInProduction, devToolsEnhancer as devToolsEnhancerLogOnlyInProduction, } from './logOnlyInProduction.js';
